src/Fmt_ast.ml

```diff
@@ -3639,14 +3639,7 @@ and fmt_structure c ctx itms =
           | _, Pstr_include {pincl_mod= me} | _, Pstr_module {pmb_expr= me} ->
               is_simple_mod me
           | _, Pstr_open _ -> true
-          | `Compact, _ ->
-              let width i =
-                let last = false in
-                String.length
-                  (Format.asprintf "%t"
-                     (Cmts.preserve (fun () -> fmt_structure_item ~last c i)))
-              in
-              width (sub_str ~ctx itm) <= c.conf.margin
+          | `Compact, _ -> Location.width itm.pstr_loc <= c.conf.margin
```

Comment:
`fmt_structure_item` lost comments even though `Cmts.preserve` is used, I don't understand why. `Location.width` looks like a good replacement, and the short-multi-liners / long-one-liners tests all pass.